src/sentry/models/outbox.py

```diff
@@ -537,23 +537,37 @@ def process_shard(
     @contextlib.contextmanager
     def process_coalesced(self) -> Generator[OutboxBase | None, None, None]:
         coalesced: OutboxBase | None = self.select_coalesced_messages().last()
+        first_coalesced: OutboxBase | None = self.select_coalesced_messages().first() or coalesced
+        tags = {"category": "None"}
+
+        if coalesced is not None:
+            tags["category"] = OutboxCategory(self.category).name
+            assert first_coalesced
```

Comment:
```suggestion
            assert first_coalesced, "Cannot get outbox queue time without two records"
```

Adding a message will make for a clearer sentry issue should this trigger.